src/libsync/owncloudpropagator.cpp

```diff
@@ -599,6 +680,16 @@ bool PropagateDirectory::scheduleNextJob()
     if (_state == NotYetStarted) {
         _state = Running;
 
+        if(_bundleJob){
+            PropagateBundle* bundle = dynamic_cast<PropagateBundle*>(_bundleJob.take());
```

Comment:
ditto  (static_cast)
Why don't you make _bundleJob  a `QScopedPointer<PropagateBundle>`  ?